### PR TITLE
feat: integrate suggestions into `flox install`

### PR DESCRIPTION
**Pull request by ysndr**
* Add a `DidYouMean` mechanism that matches terms with known package names
  and searches the top 3 packages using `pkgdb search`.
* Handle `120` ("resolution failure") errors emitted by `pkgdb manifest lock` during `install`.
* print suggestions iff **one** package was provided to install.

### CODE REVIEW

**Comment by ysndr** [comment]
Unavailable, no suggestion:
```
flox install flabbergasted
ERROR: ❌  could not install flabbergasted
```

Unavailable, searched suggestions
```
$ flox install dalle
ERROR: ❌  could not install dalle
    
    Here are a few other similar options:
      $ flox install python310Packages.dalle-mini
      $ flox install python311Packages.dalle-mini
    ...or see up to 2 options with 'flox search dalle'
```

Unavailable, curated and searched suggestions
```
$ flox install node 
ERROR: ❌  could not install node
    The package 'node' may be available as 'nodejs'.
    
    Here are a few other similar options:
      $ flox install eks-node-viewer
      $ flox install kubectl-node-shell
      $ flox install leafnode
    ...or see up to 73 options with 'flox search node'
```

**Comment by ghudgins** [comment]
changes LGTM from the product side

File: cli/flox/src/utils/didyoumean.rs
Comment (mkenigs) [inline cli/flox/src/utils/didyoumean.rs:122]: suggestion blocking: I think we're making the point about `nodejs` a bit too strongly 😆 
```
> flox install npm
ERROR: ❌  could not install npm
    Try 'flox install nodejs' instead.

    Here are a few other similar options:
      $ flox install nodejs
      $ flox install nodejs
      $ flox install nodejs
    ...or see up to 90 options with 'flox search nodejs'
```

File: cli/tests/environment-install.bats
Comment (mkenigs) [inline cli/tests/environment-install.bats:109]: nonblocking suggestion: test the multiple packages case

File: cli/flox/src/commands/environment.rs
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:915]: nit: I think it would be better to print `could not install` and then have the spinner pop up, rather than the first message be `looking for suggestions` and then printing `could not install` after that finishes

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: cli/flox/src/utils/didyoumean.rs
Comment (ysndr) [reply cli/flox/src/utils/didyoumean.rs:122]: wow ?! i have no idea what happened there

File: cli/flox/src/commands/environment.rs
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:915]: That would be a bit weird since then `Error` would be the suggestion, i.e. we would print an error outside of the error handling..
I'd like to keeo it self contained for now if possible.

Comment (mkenigs) [reply cli/flox/src/commands/environment.rs:915]: Which do you think is a better user experience?

File: cli/flox/src/utils/didyoumean.rs
Comment (mkenigs) [reply cli/flox/src/utils/didyoumean.rs:122]: Oh interesting I have multiple systems in my manifest, does that reproduce it for you?

File: cli/flox/src/commands/environment.rs
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:915]: We can make amend the spinner to say "Could not find a package for {path}, searching for suggestions"...
I think that's both clear and does not fork the error handling

Comment (mkenigs) [reply cli/flox/src/commands/environment.rs:915]: 👍 great

File: cli/flox/src/utils/didyoumean.rs
Comment (mkenigs) [reply cli/flox/src/utils/didyoumean.rs:122]: I'm not sure what needs to be fixed where, so I'm going to approve but not merge and leave it up to you

**Review by mkenigs — APPROVED** [review]

File: cli/flox/src/utils/didyoumean.rs
Comment (ysndr) [inline cli/flox/src/utils/didyoumean.rs:54]:
```suggestion
        message: &format!("Could not find package for {term}. Looking for suggestions..."),
```

File: cli/flox/src/commands/environment.rs
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:915]: added the message, needs new approval

File: cli/tests/environment-install.bats
Comment (ysndr) [reply cli/tests/environment-install.bats:109]: done

File: cli/flox/src/utils/didyoumean.rs
Comment (ysndr) [reply cli/flox/src/utils/didyoumean.rs:122]: yeah i guess that's underlying issues with search. I wouldn't touch those here but we should look at that esp. now that we add more systems to the manifest.

**Review by mkenigs — APPROVED** [review]

**Pull request merged**
